internal/command/deploy/deploy.go

```diff
@@ -477,6 +486,28 @@ func deployToMachines(
 
 	status.FlyctlVersion = buildinfo.Info().Version.String()
 
+	retriesFlag := flag.GetString(ctx, "deploy-retries")
+	deployRetries := 0
+
+	switch retriesFlag {
+	case "auto":
+		ldClient := launchdarkly.ClientFromContext(ctx)
+		retries := ldClient.GetFeatureFlagValue("deploy-retries", 0.0).(float64)
+		deployRetries = int(retries)
```

Comment:
I don't see a `launchdarkly.NewClient()` to instantiate the client. Am I missing it?

Also, there doesn't seem to be anywhere that waits for `updateFeatureFlags()` to fetch the flags before grabbing the feature flag value here. So there's probably a good chance that the flag won't be initialized by the time you get to this point in the code.

Reply:
we instantiate the ld client in the DeployWithConfig function:
https://github.com/superfly/flyctl/blob/better_recovery/internal/command/deploy/deploy.go#L301

i purposely do this before we start messing with the remote builder. that takes a few seconds at minimum, so by the time we try to get the feature flag from launchdarkly, it's very likely that the flag will be instantiated. I also confirmed this with some testing. While there's a chance that the flag isn't fetched from LD if the user's internet is being weird, I don't want to slow down deploys too much.

Would you prefer if we added a timeout option to the GetFeatureFlag function?

Reply:
I think I didn't have an updated view of the code locally. Sorry about that. I see it now.

As for the fetch, I think that having the remote builder step in between is good enough. No need to add a timeout.